File: sorting-snippets/bubblesort/bubblesort.ts
function bubblesort(array: any[]): void {
  for (let i: number = 0; i < array.length; i++) {
    let swapped: boolean = false;
    for (let j: number = 0; j < array.length-i; j++) {
      if (array[j] > array[j+1]) {
        swapped = true;
        const tmp: any = array[j];
        array[j] = array[j+1];
        array[j+1] = tmp;
      }
    }
    if (!swapped) {break;}
  }
}

const array: number[] = [1,4,2,6,2,7,2,2,4,2];
bubblesort(array);
console.log(array);

File: sorting-snippets/bubblesort/BubbleSort.java
import java.util.Arrays;

public class BubbleSort {

    public static void main(String[] args) {
        int[] array = {1,3,1,6,1,6,1,51};
        bubbleSort(array);
        System.out.println(Arrays.toString(array));
    }

    public static void bubbleSort(int[] array) {
        int len = array.length;
        for (int i = 0; i < len; i++) {
            Boolean swapped = false;
            for (int j = 0; j < len-i-1; j++) {
                if (array[j] > array[j+1]) {
                    swapped = true;
                    int tmp = array[j];
                    array[j] = array[j+1];
                    array[j+1] = tmp;
                }
            }
            if (!swapped) break;
        }
    }
}